util/telemetry/src/lib.rs

```diff
@@ -52,6 +51,14 @@ pub fn versioned_tracer(
     tracer_provider().versioned_tracer(name, version, schema_url)
 }
 
+pub fn create_async_context<T>(tracer: &BoxedTracer, name: T) -> Context
```

Comment:
What is async about this?

Reply:
The method itself is not `async` but it creates contexts for tracing inside of `async` methods. I can rename to make it clearer? Maybe just `create_context` and I add a doc comment explaining its usage inside async?

Reply:
Yeah that sounds good (renaming + documenting), thank you!

Reply:
Done.